catalog/spatial/wps/spatial-wps-endpoint/src/main/java/org/codice/ddf/spatial/ogc/wps/process/endpoint/WpsObjectConverter.java

```diff
@@ -0,0 +1,819 @@
+                .getEnumeratedValues()
+                .stream()
+                .map(
+                    ev ->
+                        createValue(
+                            ev,
+                            literalDataDescription.getType(),
+                            literalDataDescription.getUnitOfMeasure()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        allowedValues.setValueOrRange(valueTypes);
+        literalDataDomain.setAllowedValues(allowedValues);
+      } else if (literalDataDescription.getRange() != null) {
+        Range range = literalDataDescription.getRange();
+        AllowedValues allowedValues = owsObjectFactory.createAllowedValues();
+        RangeType rangeType = owsObjectFactory.createRangeType();
+        rangeType.setMaximumValue(
+            createValue(
+                range.getMaximumValue(),
+                literalDataDescription.getType(),
+                literalDataDescription.getUnitOfMeasure()));
+        rangeType.setMinimumValue(
+            createValue(
+                range.getMinimumValue(),
+                literalDataDescription.getType(),
+                literalDataDescription.getUnitOfMeasure()));
+        rangeType.setSpacing(
+            createValue(
+                range.getSpacing(),
+                literalDataDescription.getType(),
+                literalDataDescription.getUnitOfMeasure()));
+        rangeType.setRangeClosure(
+            Collections.singletonList(createRangeClosure(range.getClosure())));
+        allowedValues.setValueOrRange(Collections.singletonList(rangeType));
+        literalDataDomain.setAllowedValues(allowedValues);
+      } else if (literalDataDescription.getReference() != null) {
+        ValuesReference valuesReference = owsObjectFactory.createValuesReference();
+        valuesReference.setReference(
+            String.valueOf(literalDataDescription.getReference().getUrl()));
+        valuesReference.setValue(literalDataDescription.getReference().getDescription());
+
+        literalDataDomain.setValuesReference(valuesReference);
+
+      } else {
+        literalDataDomain.setAnyValue(owsObjectFactory.createAnyValue());
+      }
+      literalDataType.setFormat(formats);
+      literalDataType.setLiteralDataDomain(Collections.singletonList(literalDataDomain));
+      return wpsObjectFactory.createLiteralData(literalDataType);
+    } else {
+      ComplexDataType complexDataType = wpsObjectFactory.createComplexDataType();
+      complexDataType.setFormat(formats);
+      return wpsObjectFactory.createComplexData(complexDataType);
+    }
+  }
+
+  private DomainMetadataType createDataType(DataType type) {
+    DomainMetadataType domainMetadataType = owsObjectFactory.createDomainMetadataType();
+    domainMetadataType.setReference(type.reference());
+    domainMetadataType.setValue(type.value());
+    return domainMetadataType;
+  }
+
+  public String createRangeClosure(RangeClosure closure) {
```

Comment:
❓ Why have this conversion logic in this endpoint instead of inside the `RangeClosure` Enum ?